src/flexible_type/flexible_type.hpp

```diff
@@ -445,6 +445,33 @@ class flexible_type {
   template <typename T>
   const T& get() const;
 
+
+  /**
```

Comment:
This part doesn't look related to making TS_ASSERT thread safe. Should there be two separate commits here?

Reply:
Ah yes. indeed. This also fixes a bunch of other test failures. I can rename the PR. Or I can submit that as a different PR. What do you prefer?

Reply:
I think it I will split into a different PR.